services/bots.py

```diff
@@ -13,15 +14,46 @@
 log = logging.getLogger(__name__)
 
 
+def get_owner_installation_id(
+    owner: Owner,
+    deprecated_using_integration: bool,
+    *,
+    repository: Optional[Repository] = None,
+    installation_name: str = GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+) -> Optional[int]:
+    default_app_installation_filter = filter(
+        lambda obj: obj.name == installation_name, owner.github_app_installations or []
+    )
+    # filter is an Iterator, so we need to scan matches
+    # in practice we only consider the 1st one
+    for app_installation in default_app_installation_filter:
+        if repository:
+            if app_installation.is_repo_covered_by_integration(repository):
+                return app_installation.installation_id
+            # The repo we want to get a token for is not covered by the installation
+            return None
+        else:
+            # Getting owner installation - not tied to any particular repo
+            return app_installation.installation_id
+    # DEPRECATED FLOW - begin
+    if owner.integration_id and deprecated_using_integration:
```

Comment:
Could this code run? How do customers with their integration id tied to their owner start using the new GH App class? Would that be "auto-populated" after syncing?

Reply:
Yes, the code can run if there's no `GithubAppInstallation` object for a given owner.

The `GH App` objects are populated by the API when we receive the installation webhook.
I don't think there are plans for backfilling this info, so it would only be filled for an existing user if they make changes to their installation.